This notebook is created to test branching in git. I'm testing branching because I forgot to push my commit earlier today when I created the notebooks and want to avoid causing a conflict from a different PC. Hopefully this doesn't cause more hassle...

### Lady tasting tea experiment
The task is to firstly calculate the probability of choosing correctly when the lady tasting tea experiment is expanded to 12 cups total (6 milk first).

If the original has probability calculated by finding the number of combinations of ways to choose 4 cups from the possible 8, the altered version should be calculated using the same process with 12 and 6.


In [1]:
import math

In [4]:
total_cups = 12
milk_cups = 6

cups_combinations = math.comb(total_cups, milk_cups)

print(f'There are {cups_combinations} combinations.')

There are 924 combinations.


The above calculation show that there are 924 possible combinations of 6 cups that can be chosen from the total of 12. This would imply that if the lady (or other participant) is choosing at random they have a 1 in 924 chance of choosing the correct set of cups.

### Single error allowed

If the taster is allowed to make a single error then they are only required to choose 5 correct cups.

In [7]:
total_cups = 12
milk_cups = 5
cups_combinations = math.comb(total_cups, milk_cups)
cups_combinations

792